README.md

```diff
@@ -1,189 +1,237 @@
-region and partition. The task `make_plan_distrib` is a
-`__demand(__inline)` task that internally performs an index launch
-over the machine to initialize `p`.
+`make_plan` is a `__demand(__inline)` task. This means that if the user wants it
+to execute it in a separate task, they must wrap the task themselves
 
-**Important:** like `make_plan`, `make_plan_distrib` overwrites the
-input and output regions `r` and `s`.
+#### 3. Execute the plan
 
-**Important:** in order for the distributed API to work correctly, it
-is essential that each task in the index launch inside of
-`make_plan_distrib` is mapped onto a separate node. This ensures that
-when the region `p` is used later, there is a plan for every node in
-the machine.
+Next, we execute the plan. This takes the same 3 regions as mentioned above.
 
-### Execution
+```lua
+fft1d.execute_plan_task(r, s, p)
+```
 
-To perform an FFT, execute the plan. Note that, unlike with
-initialization, this looks similar whether or not distributed mode is
-being used.
+Note that `execute_plan` is a `__demand(__inline)` task (similar to `make_plan`
+above). The task `execute_plan_task` is simply a wrapper around `execute_plan`
+for convenience, to avoid needing to define this explicitly.
 
-For example, if there is a single input `r` and output `s`:
+> [!IMPORTANT]
+>
+> Because `execute_plan` is a `__demand(__inline)` task, it will never execute
+> on the GPU (unless the parent task is running on the GPU). Therefore, in most
+> cases, it is necessary to use `execute_plan_task` if one wants to use the GPU.
 
-```
-fft1d.execute_plan(r, s, p)
-```
+#### 4. Destroy the plan
 
-Or if there are partitions `r_part` and `s_part` that are distributed
-around the machine, one might do:
+When a plan is no longer needed, it can be destroyed:
 
-```
-__demand(__index_launch)
-for i in r_part.colors do
-  fft1d.execute_plan_task(r_part[i], s_part[i], p)
-end
+```lua
+fft1d.destroy_plan(p)
 ```
 
-Note that `execute_plan` is a `__demand(__inline)` task (as described
-above). The task `execute_plan_task` is simply a wrapper around
-`execute_plan` for convenience, to avoid needing to define this
-explicitly.
+### 5. Batched Transforms
 
-**Important:** because `execute_plan` is a `__demand(__inline)` task,
-it will never execute on the GPU (unless the parent task is running on
-the GPU). Therefore, in most cases it is necessary to use
-`execute_plan_task` if one wants to use the GPU.
+To illustrate how to perform a batched transform, let us use the example where you want to perform 7 batches of a 256 x 256 transform. 
 
-**Important:** while `execute_plan_task` may be executed on the GPU,
-the contents of the `p` region must still be available on the CPU,
-because the plans must be used by the host-side code to launch the FFT
-kernels. Therefore, when `execute_plan_task` is mapped onto the GPU it
-is very important to map the `p` region into zero-copy memory.
+In this case, the user creates a 3D interface:
```

Comment:
A 2D batch transform requires input/output regions of int3d since the last dimension is the number of batches.  I've thus termed the interfaces '3D'. But happy to rethink this terminology